cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TemplateUtils.java

```diff
@@ -4,23 +4,45 @@
  */
 package io.strimzi.operator.cluster.model;
 
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.strimzi.api.kafka.model.common.template.AdditionalVolume;
+import io.strimzi.api.kafka.model.common.template.ContainerTemplate;
 import io.strimzi.api.kafka.model.common.template.DeploymentStrategy;
 import io.strimzi.api.kafka.model.common.template.DeploymentTemplate;
 import io.strimzi.api.kafka.model.common.template.HasMetadataTemplate;
+import io.strimzi.api.kafka.model.common.template.PodTemplate;
+import io.strimzi.operator.common.model.InvalidResourceException;
 
+import java.util.List;
 import java.util.Map;
+import java.util.regex.Pattern;
 
 /**
  * Shared methods for working with Strimzi API templates
  */
 public class TemplateUtils {
+    /**
+     * This is a constant that represents an allowed mountable path in the file system.
+     * It is used to prevent the creation of volumes outside this path.
+     */
+    protected static final String ALLOWED_MOUNT_PATH = "/mnt";
+    /**
+     * This Pattern defines a regex for validating volume names with the following criteria:
+     * Length: Must contain at most 63 characters.
+     * Characters: Can only contain lowercase alphanumeric characters, '-', '.', or '_'.
+     * Start: Must start with an alphanumeric character.
+     * End: Must end with an alphanumeric character.
+     **/
+    public final static Pattern VOLUME_NAME_REGEX = Pattern.compile("^(?=.{0,63}$)[a-zA-Z0-9][a-zA-Z0-9-._]*[a-zA-Z0-9]$");
```

Comment:
Does it need to be public? Would this be sufficient as it seems to be used by the test only?

```suggestion
    /* test */ final static Pattern VOLUME_NAME_REGEX = Pattern.compile("^(?=.{0,63}$)[a-zA-Z0-9][a-zA-Z0-9-._]*[a-zA-Z0-9]$");
```

Or did I missed some use somewhere else?